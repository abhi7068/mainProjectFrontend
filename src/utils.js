import React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import thunk from 'redux-thunk';
import PropTypes from 'prop-types';
import reducer from '../src/store/reducers/rootReducer';
import { Router } from 'react-router-dom';

function render(
  ui,
  {
    initialState,
    store = createStore(reducer, initialState, applyMiddleware(thunk)),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    const history = createMemoryHistory();
    return (
      <Provider store={store}>
        <Router history={history}>{children}</Router>
      </Provider>
    );
  }
  Wrapper.propTypes = {
    children: PropTypes.elementType,
  };
  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
export * from '@testing-library/react';
export { render };
