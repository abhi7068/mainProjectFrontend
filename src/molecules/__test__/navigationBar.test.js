import React from 'react';
import { render, cleanup } from '../../utils';
import NavigationBar from '../Navigationbar';

afterEach(cleanup);

it('creatingsnapshot', () => {
  const { asFragment } = render(<NavigationBar />);
  expect(asFragment()).toMatchSnapshot();
});
