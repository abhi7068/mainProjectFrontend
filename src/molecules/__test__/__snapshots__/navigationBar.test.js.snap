// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`creatingsnapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-coloring-5"
  >
     
    <nav
      class="MuiToolbar-root MuiToolbar-dense makeStyles-toolbarSecondary-3 MuiToolbar-gutters"
    >
      <div>
        <div
          class="MuiFormControl-root makeStyles-formControl-11"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
            data-shrink="false"
            id="demo-controlled-open-select-label"
            style="color: black;"
          >
            Sort
          </label>
          <div
            class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="demo-controlled-open-select-label demo-controlled-open-select"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
              id="demo-controlled-open-select"
              role="button"
              tabindex="0"
            >
              <span>
                â€‹
              </span>
            </div>
            <input
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
